code/game/machinery/computer/dna_console.dm

```diff
@@ -1851,6 +1851,9 @@
 	if(stored_research && !(path in stored_research.discovered_mutations))
 		var/datum/mutation/human/HM = GET_INITIALIZED_MUTATION(path)
 		stored_research.discovered_mutations += path
+		if(HM.discovery_points)
+			stored_research.add_point_type(TECHWEB_POINT_TYPE_DISCOVERY, HM.discovery_points)
+			say("[HM.discovery_points] discovery research gained.")
```

Comment:
This should use the component, otherwise you can repeatedly get these over and over again by just resetting the stored_research

Reply:
ah that makes sense. working on it now

Reply:
I was hoping to get the points without a discovery scanner, but the discovery_scan proc in discoverable_component only happens after clicking with a discovery scanner.  I don't want to touch discoverable_component.dm if I don't have to, so I am looking for a way to do that.  Unless it's okay for me to add a signal or something for when clicking on a dna console.  Any pointer would be great.

Reply:
Add a discovery_unlock signal or something

Reply:
I'm stuck with how to handle mutations having a discoverable component.  The discoverable component early returns if thing getting the component isn't an atom.  It looks to me the whole discoverable component would have to be reworked to allow datums, and someone else told me that components aren't even supposed to be used for datums.  Let me know if doing any of that is a problem.

Also I'm not sure how resetting stored research resetting is a problem.  The dna console gets the science techweb from SSresearch.science_tech when initialized, so it seems that would have to be reset or something, I'm not really sure at this point.

I tried making a signal too, but the other problems are getting in the way.

Reply:
You could use the global list used in the discovery scanner and just check it here